### DISCUSSION TITLE
How to get a backup of filesystem volume with Kopia CLI?

### DISCUSSION THREAD

**Discussion by krzwiatrzyk** [vmware-tanzu/velero#8352] [Community support Q&A]
Like in the title, I have a backup of a volume that was mounted to a pod. This backup is in Minio storage. I can properly restore this volume with Velero restore feature.

However, I want to ensure that in some misterious cases I will be able to work on a volume backup. As I know that filesystem backup is performed via Kopia, I have installed Kopia CLI and pointed to S3 bucket (and path) where my volume is.

And:
1. `kopia repository connect s3` <-- success
2. `kopia snapshot list --all` <-- empty output
3. `kopia contents size` <-- 0B

And I wonder, doesn't using some custom kopia code break integration with Kopia CLI directly? Is it event possible anymore to get/decrypt that thata without restoring it into pod filesystem?

**Comment by kaovilai** [comment]
Velero do use a static password on kopia backups in addition, check bsl bucket prefix is added to kopia path.

**Reply by krzwiatrzyk** [reply]
It is, I connect to bucket with that command:
```
      kopia repository connect s3
      --access-key "ACCESS_KEY_ID"
      --secret-access-key "SECRET_ACCESS_KEY"
      --disable-tls
      --endpoint=<minio-address>
      --bucket=backups
      --prefix /kopia/maypp/
      --password=$PASSWORD
```

And this command is successful, If I would provider incorrect prefix than there is an error that in that path there is no repository

**Reply by kaovilai** [reply]
Try [this](https://github.com/vmware-tanzu/velero/issues/8019#issuecomment-2232926191)

> Since you have connected to the kopia repo, could you run `kopia repo status` `kopia snapshot list --all` `kopia content stats`, and share the outputs?
